telemetry/telemetrybuffer.go

```diff
@@ -464,3 +466,38 @@ func ReadConfigFile(filePath string) (TelemetryConfig, error) {
 
 	return config, err
 }
+
+// ConnectToTelemetryService - Attempt to spawn telemetry process if it's not already running.
+func (tb *TelemetryBuffer) ConnectToTelemetryService(telemetryNumRetries, telemetryWaitTimeInMilliseconds int) {
+	path, args := getTelemetryServiceDirectory()
+	for attempt := 0; attempt < 2; attempt++ {
+		if err := tb.Connect(); err != nil {
+			log.Printf("Connection to telemetry socket failed: %v", err)
+			tb.Cleanup(FdName)
+			StartTelemetryService(path, args)
+			WaitForTelemetrySocket(telemetryNumRetries, time.Duration(telemetryWaitTimeInMilliseconds))
+		} else {
+			tb.Connected = true
+			log.Printf("Connected to telemetry service")
+			return
+		}
+	}
+}
+
+func getTelemetryServiceDirectory() (path string, args []string) {
```

Comment:
this function should just return directory path, construct args in ConnectToTelemetryService function itself

Reply:
done